mmcv/ops/csrc/common/cuda/nattenav_cuda_kernel.cuh

```diff
@@ -0,0 +1,148 @@
+/*!
+**************************************************************************************************
+* NATTEN-COMMON FUNCTIONS (CUDA)
+* Copyright (c) 2020 SenseTime. All Rights Reserved.
+* Licensed under the Apache License, Version 2.0 [see LICENSE for details]
+**************************************************************************************************
+* Modified from
+*https://github.com/SHI-Labs/Neighborhood-Attention-Transformer
+**************************************************************************************************
+*/
+#include <ATen/ATen.h>
+#include <ATen/cuda/CUDAContext.h>
+#include <cuda.h>
+#include <cuda_runtime.h>
+
+#define AT_DISPATCH_HALF_TYPES(SCALARTYPE1, TYPE, NAME, ...)                \
+  [&] {                                                                     \
+    const auto& the_type = TYPE;                                            \
+    /* don't use TYPE again in case it is an expensive or side-effect op */ \
+    at::ScalarType _st = ::detail::scalar_type(the_type);                   \
+    RECORD_KERNEL_FUNCTION_DTYPE(NAME, _st);                                \
+    switch (_st) {                                                          \
+      AT_PRIVATE_CASE_TYPE(                                                 \
+          NAME, SCALARTYPE1,                                                \
+          decltype(c10::impl::ScalarTypeToCPPType<SCALARTYPE1>::t),         \
+          __VA_ARGS__)                                                      \
+      default:                                                              \
+        break;                                                              \
+    }                                                                       \
+  }()
+
+#define CUDA_NUM_THREADS 1024
+
+// CUDA: number of blocks for threads.
+inline int GET_BLOCKS(const int64_t N,
```

Comment:
We have a similar function in `common/cuda/common_cuda_helper.hpp`.

Reply:
This function has been removed.